toolbox/main.go

```diff
@@ -0,0 +1,160 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+	"time"
+
+	"istio.io/istio/tests/e2e/util"
+)
+
+var (
+	repo = flag.String("repo", "", "Update dependencies of only this repository")
+)
+
+func remote(repo string) (string, error) {
+	acct := *owner // passed in as flag in githubClient
+	token, err := getToken()
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("https://%s:%s@github.com/%s/%s.git", acct, token, acct, repo), nil
+}
+
+func replaceCommit(line string, dep dependency) (string, error) {
+	remoteURL, err := remote(dep.repoName)
+	if err != nil {
+		return line, err
+	}
+	latestStableCommit, err := util.Shell(
+		fmt.Sprintf("git ls-remote %s %s", remoteURL, dep.prodBranch))
+	if err != nil {
+		return line, err
+	}
+	commitSHA := latestStableCommit[:strings.Index(latestStableCommit, "\t")]
+	idx := strings.Index(line, "\"")
+	return line[:idx] + "\"" + commitSHA + "\",", nil
+}
+
+func updateDepFile(repo string) error {
+	rinfo := repoInfoStore[repo]
+	input, err := ioutil.ReadFile(rinfo.depFile)
+	if err != nil {
+		return err
+	}
+	lines := strings.Split(string(input), "\n")
+	for i := 0; i < len(lines); i++ {
+		for _, dep := range rinfo.deps {
+			switch dep.depType {
+			case bazelRule:
+				if strings.Contains(lines[i], fmt.Sprintf("name = \"%s\"", dep.name)) {
+					found := false
+					for ; i < len(lines) && !strings.Contains(lines[i], ")"); i++ {
+						if strings.Contains(lines[i], "commit = ") {
+							if lines[i], err = replaceCommit(lines[i], dep); err != nil {
+								return err
+							}
+							found = true
+						}
+					}
+					if !found {
+						return fmt.Errorf("Failure on repo %s: 'commit' is not defined for %s", repo, dep.name)
+					}
+				}
+			case constant:
+				if strings.Contains(lines[i], dep.name) {
+					if lines[i], err = replaceCommit(lines[i], dep); err != nil {
+						return err
+					}
+				}
+			}
+		}
+	}
+	output := strings.Join(lines, "\n")
+	return ioutil.WriteFile(rinfo.depFile, []byte(output), 0600)
+}
+
+func updateDeps(repo string) error {
+	if _, err := util.Shell("rm -rf " + repo); err != nil {
+		return err
+	}
+	remoteURL, err := remote(repo)
+	if err != nil {
+		return err
+	}
+	if _, err := util.Shell("git clone " + remoteURL); err != nil {
+		return err
+	}
+	if err := os.Chdir(repo); err != nil {
+		return err
+	}
+	branch := fmt.Sprintf("autoUpdateDeps%v", time.Now().UnixNano())
```

Comment:
That s unique, but that does not prevent you from creating twice the same PR. if should be a function of (Parent Repo, Deps Repos, Branch Name, Head of Branch, SHAs to update)